### PR TITLE
Upgrade to MUX 2.8

### LINKED ISSUE #13495: Update to WinUI 2.8 (was: History size overlaps with widget controls)
### Windows Terminal version

1.15.1863.0

### Windows build number

10.0.22000.778

### Other Software

_No response_

### Steps to reproduce

Enter a large history size, INT_MAX used for testing

### Expected Behavior

No overlapping

### Actual Behavior

max_history_size

## _maintainer note_

Also:
* [ ] #9822

### PR DESCRIPTION
**Pull request by zadjii-msft**
Updates the Terminal to Microsoft.UI.Xaml v2.8. 

* MUX 2.8 adds a dependency on WebView2, so we need to include that too.
* See https://github.com/microsoft/microsoft-ui-xaml/pull/7574 for why we're adding the `.props`
* The TabView thing: 
  > tl;dr: In >=MUX 2.7, we were updating our tab colors by doing a "Visual State Dance", as I called it. We'd manually change the `TabViewItem`'s VisualState to one that it wasn't in, then change it back to the one it should be in. This seemingly re-applied the new values of the brushes. However in 2.8, this seemingly didn't work anymore!
  > 
  > So instead, we do a "Theme Dance", like so:
  > ```c++
  >   const auto& reqTheme = TabViewItem().RequestedTheme();
  >   TabViewItem().RequestedTheme(ElementTheme::Light);
  >   TabViewItem().RequestedTheme(ElementTheme::Dark);
  >   TabViewItem().RequestedTheme(reqTheme);
  >   ```
  > This causes the `ThemeResource`s to be re-evaluated to the new values. 
  > We never got to the root cause of why this seems different in 2.8. It literally makes no sense. 

### checklist 

* [x] closes #13495

### CODE REVIEW

File: src/cascadia/TerminalApp/TabBase.cpp [hunk 1]
```diff
@@ -556,24 +561,19 @@ namespace winrt::TerminalApp::implementation
     }
 
     // Method Description:
-    // Toggles the visual state of the tab view item,
-    // so that changes to the tab color are reflected immediately
+    // - Toggles the requested theme of the tab view item,
```
Comment (zadjii-msft) [inline src/cascadia/TerminalApp/TabBase.cpp:564]:
```suggestion
    // BODGY
    // - Toggles the requested theme of the tab view item,
```

**Review by carlos-zamora — APPROVED** [review]

File: src/cascadia/TerminalApp/TabBase.cpp [hunk 2]
```diff
@@ -556,24 +561,20 @@ namespace winrt::TerminalApp::implementation
...
-        }
-        else
-        {
-            VisualStateManager::GoToState(TabViewItem(), L"Selected", true);
-            VisualStateManager::GoToState(TabViewItem(), L"Normal", true);
-        }
+        const auto& reqTheme = TabViewItem().RequestedTheme();
+        TabViewItem().RequestedTheme(ElementTheme::Light);
+        TabViewItem().RequestedTheme(ElementTheme::Dark);
+        TabViewItem().RequestedTheme(reqTheme);
```
Comment (lhecker) [inline src/cascadia/TerminalApp/TabBase.cpp:578]: BODGY comment pending. 😄

**Review by lhecker — APPROVED** [review]

**Comment by DHowett** [comment]
Make sure you clean up the PR body!

File: dep/nuget/packages.config [hunk 3]
```diff
@@ -8,7 +8,8 @@
   <package id="Microsoft.VCRTForwarders.140" version="1.0.4" targetFramework="native" />
   <package id="Microsoft.Internal.Windows.Terminal.ThemeHelpers" version="0.6.220404001" targetFramework="native" />
   <package id="Microsoft.VisualStudio.Setup.Configuration.Native" version="2.3.2262" targetFramework="native" developmentDependency="true" />
-  <package id="Microsoft.UI.Xaml" version="2.7.3" targetFramework="native" />
+  <package id="Microsoft.UI.Xaml" version="2.8.2" targetFramework="native" />
+  <package id="Microsoft.Web.WebView2" version="1.0.1661.34" targetFramework="native" />
```
Comment (DHowett) [inline dep/nuget/packages.config:12]: TODO Dustin: look at the content of this package and see if we can suppress the DLL being copied in.

**Comment by zadjii-msft** [comment]
@DHowett Should we block while we wait on the WebView2 thing?

**Review by DHowett — CHANGES_REQUESTED** [review]
just for a little.

File: dep/nuget/packages.config (hunk 3 above)
Comment (DHowett) [reply dep/nuget/packages.config:12]: Looked at, fixed.

File: src/cascadia/UnitTests_Control/Control.UnitTests.vcxproj [hunk 4]
```diff
@@ -13,6 +13,12 @@
   <PropertyGroup Label="NuGet Dependencies">
     <TerminalCppWinrt>true</TerminalCppWinrt>
     <TerminalMUX>true</TerminalMUX>
+
+    <!-- This magic property will let WebView2 know that it needs to reveal its
+    winmd to us. Without it, the WebView2 targets don't think we're a WinRT
+    context, and our build will fail when MUX fails midl compile (due to missing
+    webview components). -->
+    <WebView2UseWinRT>true</WebView2UseWinRT>
```
Comment (DHowett) [inline src/cascadia/UnitTests_Control/Control.UnitTests.vcxproj:21]:
```suggestion
```

File: src/cascadia/WindowsTerminal/WindowsTerminal.vcxproj [hunk 5]
```diff
@@ -27,6 +27,14 @@
...
   <Import Project="$(OpenConsoleDir)src\cppwinrt.build.pre.props" />
 
+  <PropertyGroup>
+    <!-- This magic property will let WebView2 know that it needs to reveal its
+    winmd to us. Without it, the WebView2 targets don't think we're a WinRT
+    context, and our build will fail when MUX fails midl compile (due to missing
+    webview components). -->
+    <WebView2UseWinRT>true</WebView2UseWinRT>
+  </PropertyGroup>
+
```
Comment (DHowett) [inline src/cascadia/WindowsTerminal/WindowsTerminal.vcxproj:37]:
```suggestion
```

**Pull request merged**